src/main/java/com/gmail/nossr50/skills/herbalism/HerbalismManager.java

```diff
@@ -767,60 +765,63 @@ private boolean processGreenThumbPlants(BlockState blockState, BlockBreakEvent b
             case "carrots":
                 seed = Material.matchMaterial("CARROT");
                 break;
-
             case "wheat":
                 seed = Material.matchMaterial("WHEAT_SEEDS");
                 break;
-
             case "nether_wart":
-                seed = Material.getMaterial("NETHER_WART");
+                seed = Material.matchMaterial("NETHER_WART");
                 break;
-
             case "potatoes":
                 seed = Material.matchMaterial("POTATO");
                 break;
-
             case "beetroots":
                 seed = Material.matchMaterial("BEETROOT_SEEDS");
                 break;
-
             case "cocoa":
                 seed = Material.matchMaterial("COCOA_BEANS");
                 break;
-
             case "torchflower":
                 seed = Material.matchMaterial("TORCHFLOWER_SEEDS");
                 break;
             default:
                 return false;
         }
 
-
         ItemStack seedStack = new ItemStack(seed);
 
         if (ItemUtils.isAxe(blockBreakEvent.getPlayer().getInventory().getItemInMainHand())
-        && blockState.getType() != Material.COCOA) {
+                && blockState.getType() != Material.COCOA) {
             return false;
         }
 
         if (!greenTerra && !ProbabilityUtil.isSkillRNGSuccessful(SubSkillType.HERBALISM_GREEN_THUMB, player)) {
             return false;
         }
 
-        if (!playerInventory.containsAtLeast(seedStack, 1)) {
+        // Check for seeds in both the main hand and off-hand
+        if (!(playerInventory.containsAtLeast(seedStack, 1) || playerInventory.getItemInOffHand().getType() == seed)) {
+            NotificationManager.sendPlayerInformation(player, NotificationType.REQUIREMENTS_NOT_MET, "Skills.NeedMore", StringUtils.getPrettyItemString(seed));
             return false;
         }
 
         if (!processGrowingPlants(blockState, ageable, blockBreakEvent, greenTerra)) {
             return false;
         }
 
-        if(EventUtils.callSubSkillBlockEvent(player, SubSkillType.HERBALISM_GREEN_THUMB, blockState.getBlock()).isCancelled()) {
+        // Check the SubSkillEvent before removing items
+        if (EventUtils.callSubSkillBlockEvent(player, SubSkillType.HERBALISM_GREEN_THUMB, blockState.getBlock()).isCancelled()) {
             return false;
         } else {
-            playerInventory.removeItem(seedStack);
+            // Remove seed from the appropriate inventory slot (main hand or off-hand)
+            if (playerInventory.getItemInMainHand().getType() == seed) {
```

Comment:
These changes are currently only checking for the item being in their main hand or offhand, this doesn't account for the seeds being in their inventory